### PR TITLE
[daemon] let daemon choose mount target if not specified

### PR DESCRIPTION
**Pull request by andrei-toterman**
This PR adds the functionality for the daemon to choose a mount target path if it is not given. Right now the logic is to take the last directory name in the source path and append it to `/home/ubuntu` e.g. given the source path `/home/andrei/projects/multipass`, the default target path would be `/home/ubuntu/multipass`

### CODE REVIEW

File: include/multipass/utils.h
Comment (ricab) [inline include/multipass/utils.h:239]: Could we have a shorter name, please? C++ is verbose enough as it is, and source is already mentioned in the parameter, so perhaps just:

```suggestion
    virtual Path default_mount_target(const Path& source) const;
```

File: src/utils/utils.cpp
Comment (ricab) [inline src/utils/utils.cpp:587]: Shouldn't we throw if this is empty? You probably want to do that in this function's client (daemon) and make it a pre-condition here that the directory isn't empty, perhaps with an assert.

**Review by ricab — COMMENTED** [review]
Thanks Andrei, just a quick review pass.

File: src/utils/utils.cpp
Comment (andrei-toterman) [reply src/utils/utils.cpp:587]: There is a purpose for returning an empty path if given an empty path, which is the GUI target preview. If the source field in the GUI is left empty, the target should also be empty. So this just simplifies that process. But I understand that taken in isolation, the better behavior for this method is to error if the source is empty.

File: include/multipass/utils.h
Comment (andrei-toterman) [reply include/multipass/utils.h:239]: Sure, makes sense. I'll change that everywhere.

File: src/utils/utils.cpp
Comment (ricab) [reply src/utils/utils.cpp:587]: Hmm, well that could be done in the GUI itself, no? Anyway, I am OK keeping it like this as long as the daemon itself handles the case.

**Review by ricab — APPROVED** [review]
Thanks @andrei-toterman! Let's wait for a primary reviewer, but LGTM on secondary review.

**Review by Sploder12 — APPROVED** [review]
LGTM! `default_mount_target` returning `""` is handled by the `invalid_target_path` check a little later so that edge case is properly handled by the daemon.

**Review by ricab — APPROVED** [review]
Alright, this is going in, thanks!

**Pull request merged**
